programs/gateway/src/lib.rs

```diff
@@ -441,6 +441,24 @@ pub mod gateway {
         Ok(())
     }
 
+    /// Updates the PDA nonce. Caller is authority stored in PDA.
+    ///
+    /// # Arguments
+    /// * `ctx` - The instruction context.
+    /// * `new_nonce` - The new nonce.
+    pub fn update_nonce(ctx: Context<UpdateNonce>, new_nonce: u64) -> Result<()> {
```

Comment:
Can we call it reset_nonce to keepl consistency with Sui and the message we have in the blockchain? 

Idea is that you reset it to a value for normal operation again